Vesper-Agents-Suite/Compound-Collateral-Factor-Agent/src/agent.ts

```diff
@@ -0,0 +1,45 @@
+import {
+  Finding, FindingSeverity,
+  FindingType, HandleTransaction,
+  TransactionEvent
+} from 'forta-agent';
+import {
+  decodeParameters,
+  FindingGenerator,
+  provideEventCheckerHandler
+} from 'forta-agent-tools';
+import { COLLATERAL_FACTOR_EVENT_ALERT_ID, COMPOUND_COMPTROLLER_ADDRESS, NEW_COLLATERAL_FACTOR_SIGNATURE } from './utils';
+
+const createFindingGenerator = (alertId = COLLATERAL_FACTOR_EVENT_ALERT_ID): FindingGenerator => {
+  return (metadata: { [key: string]: any } | undefined) => {
+    const { 2: newCollateralFactorMantissa } = decodeParameters(
+      ['address', 'uint256', 'uint256'],
+      metadata?.data,
+    );
+    return Finding.fromObject({
+      name: 'COMPOUND NEW COLLATERAL FACTOR EVENT',
+      description: 'Updated collateral factor mantissa for Compound',
+      alertId,
+      type: FindingType.Info,
+      severity: FindingSeverity.Info,
+      protocol: 'Compound',
+      metadata: {
+        mantissa: newCollateralFactorMantissa,
+      }
+    })
+  }
+}
+
+const handleTransaction: HandleTransaction = async (txEvent: TransactionEvent) => {
+  const agentHandler = provideEventCheckerHandler(
+    createFindingGenerator(),
+    NEW_COLLATERAL_FACTOR_SIGNATURE,
+    COMPOUND_COMPTROLLER_ADDRESS,
+  );
+
+  return agentHandler(txEvent);
+}
```

Comment:
I think that you can check this function a bit in order to avoid the creation of the `agentHandler` each time that the agent receives a txn.

Options are:
- Add a provider:
```typescript
const provideHandleTransaction = (alertId) => provideEventCheckerHandler(...);
```
- Use the `provideEventCheckerHandler` to create the handler directly
```typescript
const handleTransaction: HandleTransaction = provideEventCheckerHandler(...)
```